pkgs/development/interpreters/ueforth/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, stdenv, fetchFromGitHub
+, nodejs, python
+}:
+let
+  REVISION = "830e6b40d7d3a4c67302c65359c4989e2ff5a5de";
+in stdenv.mkDerivation {
+  name = "ueforth";
+  inherit REVISION;
+  src = fetchFromGitHub {
+    owner = "flagxor";
+    repo = "eforth";
+    rev = REVISION;
+    sha256 = "0xc5s3vyq3h5xjkvdxrq3fklg6pq1xsq7gy1p7aih937x4vcrlgh";
+  };
+
+  postUnpack = ''
+    sourceRoot=$sourceRoot/ueforth
+  '';
+
+  patchPhase = ''
+    sed -i 's_/usr/bin/env nodejs_${nodejs}/bin/node_' **/*.js
+    sed -i 's_/usr/bin/env python_${python}/bin/python_' **/*.py
```

Comment:
Please use `substituteInPlace`

Reply:
Hm so the alternative would be something like

```sh
find . -name "*.js" -exec substituteInPlace {} --replace "/usr/bin/env nodejs" "..." \;
```

or even since substituteInPlace is probably a shell function

```sh
for f in $(find ...)
 do substituteInPlace ...
done
```

wouldn't that be less readable than the sed call?

Reply:
Ok, that is fine.